backend/apps/ecommerce/mutations.py

```diff
@@ -0,0 +1,154 @@
+import random
+import uuid
+
+import graphene
+from django.core.exceptions import PermissionDenied
+from django.db import transaction
+from faker import Faker
+from graphql_jwt.decorators import login_required, staff_member_required
+
+
+from ..organizations.models import Organization
+from .models import Order, Product
+from .types import OrderType, ProductType
+from .vipps_utils import VippsApi
+
+
+class InitiateOrder(graphene.Mutation):
+
+    redirect = graphene.String()
+    order_id = graphene.String()
+    vipps_api = VippsApi()
+
+    class Arguments:
+        product_id = graphene.ID(required=True)
+        quantity = graphene.Int()
+
+    @login_required
+    def mutate(self, info, product_id, quantity=1):
```

Comment:
There are multiple scenarios here where the user can encounter an exception if they try to do something unexpected, and rather than propagating that error to the user, we could catch it and simply return a state saying the request was unsuccessful.